### ISSUE TITLE
Requires `if __name__ == '__main__'`

### ISSUE THREAD

**Issue by hunse** [nengo/nengo-gui#101]
@cchan987 told me to put this at the end of my file to run the visualizer

``` python
if __name__ == '__main__':
    import nengo_viz
    nengo_viz.Viz(__file__).start()
```

but if I just put this

``` python
import nengo_viz
nengo_viz.Viz(__file__).start()
```

it doesn't work. I get: 

```
Traceback (most recent call last):
  File "test_split.py", line 39, in <module>
    nengo_viz.Viz(__file__).start()
  File "/home/ehunsber/workspace/nengo_viz/nengo_viz/viz.py", line 231, in __init__
    exec(code, locals)
  File "<string>", line 39, in <module>
NameError: name '__file__' is not defined
```

**Comment by tbekolay** [comment]
I also ran into this issue, and it's not an easy one to debug. FYI the reason is because `nengo_viz` takes the file pointed to in `__file__`, reads it into a string, and `exec`s it. `__file__` exists when you run `python my_file.py`, and is `my_file.py`, but not when the file is `exec`d.

Ideally, we would either

- Make it possible to start visualizing a network without having to re-execute that network's file (in general, it would be nice to not have to have the file requirement, e.g., for starting a visualization in an IPython notebook).
- Provide a simple script and recommend starting the visualizer by running that script; i.e., `nengo_viz my_file.py`. This can be done just by editing `setup.py`.

**Comment by tcstewar** [comment]
> Make it possible to start visualizing a network without having to re-execute that network's file (in general, it would be nice to not have to have the file requirement, e.g., for starting a visualization in an IPython notebook).

That's doable right now.  You do it this way:

```python
nengo_viz.Viz(file='whatever.py', locals=locals())
```

the only thing it uses ```file``` for in this case is to generate a file name for the ```.cfg``` file.  If you pass in ```locals``` it uses that, otherwise it executes ```file``` to generate the locals.



> Provide a simple script and recommend starting the visualizer by running that script; i.e., nengo_viz my_file.py. This can be done just by editing setup.py

That is definitely the right thing to have, and it would be exactly like how ```nengo_gui``` does it now.

One other thing that I would very much like to add is to get rid of the ```if __name__ == '__main__':``` requirement.  That would be done by adding some hooks into the code that is executing the script to have it stop if it notices a nengo_viz.Viz object being created.  It should also stop if a nengo.Simulator object is created.

**Comment by tbekolay** [comment]
> That's doable right now.

Oh excellent! That makes sense for the filename etc.

> adding some hooks into the code that is executing the script

I think rather than messing with any of the execution stuff, it'd be easier to just strip those lines out of the code that is about to be executed.

**Comment by tcstewar** [comment]
> I think rather than messing with any of the execution stuff, it'd be easier to just strip those lines out of the code that is about to be executed.

Possibly...  I'm a little worried about just parsing the text of the line and doing something like "strip out lines that include ```Viz``` or ```Simulator``` (or strip out everything from those lines and later), since I can think of lots of hard-to-parse examples that shouldn't be stripped out (like ```a = nengo.Ensemble(100, 1, label="Simulator")```).  So I was planning on using the same sort of trick as ```nengo_gui``` uses to figure out the line of code that creates each object: it monkey-patches in a bit of code into the constructor for the objects.  But I haven't really thought about the details yet.

**Comment by tbekolay** [comment]
This was fixed in #224, in that `if __name__ == '__main__'` is no longer required. Other discussions about what to do when evalling user code can be done in separate issues :rainbow:
